src/main/java/com/knightuser/rest/UserResource.java

```diff
@@ -0,0 +1,71 @@
+package com.knightuser.rest;
+
+import static com.knightuser.Constants.USERS_BASE_URL;
+import static java.util.stream.Collectors.toList;
+
+import com.knightuser.domain.UserVO;
+import com.knightuser.rest.mapper.RestUserMapper;
+import com.knightuser.rest.request.UserRequestDto;
+import com.knightuser.rest.response.UserResponseDto;
+import com.knightuser.service.UserService;
+import java.util.List;
+import javax.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Validated
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(USERS_BASE_URL)
+@Slf4j
+public class UserResource {
```

Comment:
Do you really need @Vaildated here? As I know, it is used for validating by steps

Reply:
I'm not sure but I assume that without it inner validation of request objects doesn't work. I'll check, thanks.

Reply:
I think inner validation should work without it, as it is in todo microservice